File: VB/PivotGridControl_CustomCellEdit/Form1.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraPivotGrid

Namespace PivotGridControl_CustomCellEdit
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        ' Creates new repository items.
        Dim riProgressBar As New RepositoryItemProgressBar()
        Dim riTextEdit As New RepositoryItemTextEdit()

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            salesPersonTableAdapter.Fill(Me.nwindDataSet.SalesPerson)

            ' Specifies the type of data field and format settings.
            fieldQuantity1.SummaryDisplayType = DevExpress.Data.PivotGrid.PivotSummaryDisplayType.PercentOfColumn
            fieldQuantity1.CellFormat.FormatType = DevExpress.Utils.FormatType.Custom
            fieldQuantity1.CellFormat.FormatString = "{0}%"

            ' Initializes cell editors for this Pivot Grid control.
            pivotGridControl1.RepositoryItems.AddRange(New RepositoryItem() {riProgressBar, riTextEdit})
        End Sub

        Private Sub pivotGridControl1_CustomCellEdit(ByVal sender As Object,
                            ByVal e As PivotCustomCellEditEventArgs) Handles pivotGridControl1.CustomCellEdit
            ' Specifies a cell editor which is used in both display and edit modes.
            If e.DataField Is fieldQuantity1 And e.RowValueType = PivotGridValueType.Value Then
                e.RepositoryItem = riProgressBar
            End If
        End Sub

        Private Sub pivotGridControl1_CustomCellEditForEditing(sender As Object,
                    ByVal e As PivotCustomCellEditEventArgs) Handles pivotGridControl1.CustomCellEditForEditing
            ' Overrides the cell editor for the edit mode.
            If e.DataField Is fieldQuantity1 And e.RowValueType = PivotGridValueType.Value Then
                e.RepositoryItem = riTextEdit
            End If
        End Sub

        Private Sub pivotGridControl1_CustomCellValue(ByVal sender As Object,
                            ByVal e As PivotCellValueEventArgs) Handles pivotGridControl1.CustomCellValue
            If e.DataField Is fieldQuantity1 Then
                e.Value = Math.Round(Convert.ToDecimal(e.Value) * 100, 2)
            End If
        End Sub
    End Class
End Namespace

File: CS/PivotGridControl_CustomCellEdit/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.Data.PivotGrid;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraPivotGrid;

namespace PivotGridControl_CustomCellEdit {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        // Creates new repository items.
        RepositoryItemTextEdit riTextEdit = new RepositoryItemTextEdit();
        RepositoryItemProgressBar riProgressBar = new RepositoryItemProgressBar();

        private void Form1_Load(object sender, EventArgs e) {
            this.salesPersonTableAdapter.Fill(this.nwindDataSet.SalesPerson);

            // Specifies the type of data field and format settings.
            fieldQuantityPercent.SummaryDisplayType = PivotSummaryDisplayType.PercentOfColumn;
            fieldQuantityPercent.CellFormat.FormatType = DevExpress.Utils.FormatType.Custom;
            fieldQuantityPercent.CellFormat.FormatString = "{0}%";

            // Initializes cell editors for this Pivot Grid control.
            pivotGridControl1.RepositoryItems.AddRange(new RepositoryItem[] { riTextEdit, riProgressBar });
        }

        void pivotGridControl1_CustomCellEdit(object sender, PivotCustomCellEditEventArgs e) {
            // Specifies a cell editor which is used in both display and edit modes.
            if (e.DataField == fieldQuantityPercent & e.RowValueType == PivotGridValueType.Value)
                    e.RepositoryItem = riProgressBar;            
        }

        void pivotGridControl1_CustomCellEditForEditing(object sender, PivotCustomCellEditEventArgs e) {
            // Overrides the cell editor for the edit mode.
            if (e.DataField == fieldQuantityPercent & e.RowValueType == PivotGridValueType.Value) 
                    e.RepositoryItem = riTextEdit;            
        }
        
        private void pivotGridControl1_CustomCellValue(object sender, PivotCellValueEventArgs e) {
            if (e.DataField == fieldQuantityPercent)
                e.Value = Math.Round(Convert.ToDecimal(e.Value) * 100, 2);
        }

    }
}
